[2.7.4] Autoconf: Symbol versioning check should be disabled by default, but condition is wrong

While compiling 2.7.4 on Gentoo with clang 21, I get the following error:
```
[...]
rclibtool: link: clang-21 .libs/xmlparse.o .libs/xmltok.o .libs/xmlrole.o -m32 -mfpmath=sse -Wall -Wextra -fexceptions -Wmissing-prototypes -Wstrict-prototypes -pedantic -Wnull-dereference -Wjump-misses-init -Wdouble-promotion -Wshadow -Wformat=2 -Wmisleading-indentation -fvisibility=hidden -O3 -march=native -pipe -glldb -flto=thin -fwhole-program-vtables -gline-tables-only -Wstrict-aliasing=3 -lm ,./libexpat.map -Wl,-O3,-z,now,--as-needed,--lto-O3,--icf=safe,--gc-sections,-z,pack-relative-relocs -shared -fPIC -Wl,--no-undefined -Wl,-soname -Wl,libexpat.so.1 -o .libs/libexpat.so.1.11.2
clang-21: error: no such file or directory: ',./libexpat.map'
rclibtool: error logged in slbt_exec_link_create_library(), line 369: flow error: unexpected condition or other.
rclibtool: < returned to > slbt_exec_link(), line 407.
make[2]: *** [Makefile:540: libexpat.la] Error 2
make[2]: Leaving directory '/var/tmp/portage/dev-libs/expat-2.7.4/work/expat-2.7.4-abi_x86_32.x86/lib'
make[1]: *** [Makefile:643: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/dev-libs/expat-2.7.4/work/expat-2.7.4-abi_x86_32.x86'
make: *** [Makefile:521: all] Error 2
```

## Summary

expat 2.7.4 fails to build when using slibtool due to a shell quoting bug in `configure.ac`.

## Error

```
clang-21: error: no such file or directory: ',./libexpat.map'
```

## Root Cause

In `configure.ac` lines 154-155, the `AM_CONDITIONAL` test is missing the standard `x` prefix:

```m4
AM_CONDITIONAL([HAVE_VSCRIPT],
  [test "$enable_symbol_versioning" != xno])
```

When `enable_symbol_versioning=no` (the default), the test evaluates as:
- `"no" != xno` → **TRUE** (incorrect)

Compare to line 150 which correctly uses:
```m4
AS_IF([test "x$enable_symbol_versioning" != xno],
```

This causes `HAVE_VSCRIPT_TRUE` to be set even when symbol versioning is disabled, while `VSCRIPT_LDFLAGS` remains empty. The resulting Makefile contains:

```makefile
libexpat_la_LDFLAGS += ,./libexpat.map
```

The leading comma with no preceding flag is passed to the linker, causing the build failure.

## Fix

```diff
 AM_CONDITIONAL([HAVE_VSCRIPT],
-  [test "$enable_symbol_versioning" != xno])
+  [test "x$enable_symbol_versioning" != xno])
```

## Environment

- expat 2.7.4
- slibtool (rclibtool)
- clang-21
- Gentoo Linux

## Notes

This bug was introduced in 2.7.4 with the new `--enable-symbol-versioning` feature in b9191345 . The issue does not occur with GNU libtool because it may handle the malformed argument differently, but it breaks builds using slibtool.

--- a/expat/configure.ac
+++ b/expat/configure.ac
@@ -152,7 +152,7 @@ AS_IF([test "x$enable_symbol_versioning" != xno],
    AC_SUBST([VSCRIPT_LDFLAGS])
   ])
 AM_CONDITIONAL([HAVE_VSCRIPT],
-  [test "$enable_symbol_versioning" != xno])
+  [test "x$enable_symbol_versioning" != xno])
 
 dnl patching ${archive_cmds} to affect generation of file "libtool" to fix linking with clang (issue #312)
 AS_CASE(["$LD"],[*clang*],

